README.md

```diff
@@ -5,7 +5,144 @@
 [![CodeQL](https://github.com/Mellanox/maintenance-operator/actions/workflows/codeql.yml/badge.svg)](https://github.com/Mellanox/maintenance-operator/actions/workflows/codeql.yml)
 [![Image push](https://github.com/Mellanox/maintenance-operator/actions/workflows/image-push-main.yml/badge.svg?event=push)](https://github.com/Mellanox/maintenance-operator/actions/workflows/image-push-main.yml)
 
-# Nvidia Maintenance Operator
-coordinates node maintenance operations in K8s cluster
+# NVIDIA Maintenance Operator
 
-> __NOTE__: This project is currently under active development.
+NVIDIA Maintenance Operator provides Kubernetes API(Custom Resource Definition) to allow node maintenance operators in K8s cluster
+in a coordinated manner. It performs some common operations to prepare a node for maintenance such as cordoning
+the node as well as draining it.
+
+Users/Consumers can request to perform maintenance on a node by creating NodeMaintenance Custom Resource(CR).
+The operator will then reconcile NodeMaintenance CRs. At high level this the the reconcile flow:
+
+1. Scheduling - schedule NodeMaintenance to be processed by the operator, taking into account constraints
+  such as the maximal allowed parallel operations.
+2. Node preparation for maintenance such as cordon and draning of the node
+3. Mark NodeMaintenance as Ready (via condition)
+4. Cleanup on deletion of NodeMaintenance such as node uncordon
+
+## Deployment
+
+### Prerequisites
+
+* Kubernetes cluster
+
+### Helm
+
+#### Deploy latest from project sources
```

Comment:
Do you want to add a link to the Helm chart readme to this section?

Reply:
i do.